### PR TITLE
feat: Added Steps and centralized Headers

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
Adding step markers and centralizing the headers. Before the headers were in two separate files, now they are in the main index.tsx component. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2021-06-08 at 11 07 51 AM
Screen Shot 2021-06-08 at 11 07 59 AM
Screen Shot 2021-06-09 at 5 37 30 PM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx [hunk 1]
```diff
@@ -59,25 +63,26 @@ interface FieldPropTypes {
   db?: DatabaseObject;
   isEditMode?: boolean;
   sslForced?: boolean;
+  uploadOption?: string | null;
+  setUploadOption: (obj: any) => void;
+  fileToUpload?: string;
+  setFileToUpload: (obj: any) => void;
```
Comment (hughhhh) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:69]: Remove these

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx [hunk 2]
```diff
@@ -127,17 +132,20 @@ const credentialsInfo = ({
             className="input-upload"
             type="file"
             onChange={async event => {
-              const file = event?.target?.files[0];
-              setFileToUpload(file.name);
+              let file;
+              if (event.target.files) {
+                file = event.target.files[0];
+              }
```
Comment (hughhhh) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:138]: Is this to stop typescript errors?

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/ModalHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  const noDbHeader = (
+    <StyledFormHeader>
+      <div className="select-db">
+        <p className="helper"> Step 1 of 3 </p>
+        <h4>Select a database to connect</h4>
+      </div>
+    </StyledFormHeader>
+  );
+
+  if (isLoading) return <></>;
```
Comment (hughhhh) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/ModalHeader.tsx:92]: nit: prefer a switch statement here

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/ModalHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        <a href={DOCUMENTATION_LINK} target="_blank" rel="noopener noreferrer">
+          here
+        </a>
+        .
+      </CreateHeaderSubtitle>
+    </>
+  );
+  const hasConnectedDbHeader = (
+    <StyledFormHeader>
+      <p className="helper"> Step 3 of 3 </p>
```
Comment (hughhhh) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/ModalHeader.tsx:71]: Add additional informationn from spec
https://www.figma.com/file/NbuTOMaUdv2GCBoDyihdLw/Database-Connection?node-id=1698%3A401419

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx (hunk 2 above)
Comment (AAfghahi) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:138]: yeah it is

**Pull request merged**